src/features/Header/Header.jsx

```diff
@@ -1,12 +1,86 @@
+import PropTypes from 'prop-types';
 import styles from './Header.module.scss';
 import Navbar from './components/Navbar';
 import Search from './components/Search';
+import Logo from '../../base/Logo';
+import SearchIcon from '../../base/SearchIcon';
 
-const Header = () => (
-  <div className={styles.wrapper}>
-    <Navbar />
-    <Search />
-  </div>
-);
+const Header = ({ movie, isMovieDetailsShown, toggleMovieDetails }) => {
+  const moviePosterURL = movie.poster_path
+    ? movie.poster_path
+    : require('../../../public/img/posters/image_not_found.png');
+
+  const handleImgError = (e) => {
+    e.target.src = require('../../../public/img/posters/image_not_found.png');
+  };
+
+  return (
+    <div className={styles.wrapper}>
+      {isMovieDetailsShown ? (
+        <>
+          <div className={styles.nav_wrapper}>
+            <Logo />
+            <SearchIcon toggleMovieDetails={toggleMovieDetails} />
+          </div>
+          <div className={styles.movie_detail_wrapper}>
+            <img
+              src={moviePosterURL}
+              onError={handleImgError}
+              alt={movie.title}
+              className={styles.movie_poster}
+            />
+            <div className={styles.movie_details}>
+              <h2 className={styles.movie_title_wrapper}>
+                <span>{movie.title}</span>
+                <span className={styles.movie_rating}>
+                  {movie.vote_average}
+                </span>
+              </h2>
+              <div className={styles.movie_genre}>
+                {movie.genres?.join(', ')}
```

Comment:
don't 'calculate' values in JSX
on every re-render they will be 're-calculated'
move them before 'return' and ideally use useMemo